fil-proofs-tooling/scripts/retry.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# This software is freely available under the Apache 2.0 license.
+# Full license available here: https://www.apache.org/licenses/LICENSE-2.0.txt
+
+# Inspired by https://gist.github.com/reacocard/28611bfaa2395072119464521d48729a
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Retry a command on a particular exit code, up to a max number of attempts,
+# with exponential backoff.
+# Invocation:
+#   err_retry exit_code attempts sleep_multiplier <command>
+# exit_code: The exit code to retry on.
+# attempts: The number of attempts to make.
+# sleep_millis: Multiplier for sleep between attempts. Examples:
+#     If multiplier is 1000, sleep intervals are 1, 2, 4, 8, 16, etc. seconds.
+#     If multiplier is 5000, sleep intervals are 5, 10, 20, 40, 80, etc. seconds.
+
+exit_code=$1
+attempts=$2
+sleep_millis=$3
+shift 3
+
+for attempt in `seq 1 $attempts`; do
+    # This weird construction lets us capture return codes under -o errexit
+    "$@" && rc=$? || rc=$?
```

Comment:
@laser, I see that this comes from the source gist. Do you understand what case is caught by the first part of 'this weird construction' that wouldn't be if the line above were just `rc=$?`?

Reply:
The `"$@"` executes the command, and putting everything on the same line means that we evaluate the `rc=$?` is evaluated _even if the command produced a non-zero exit code_. Given that I'm using `set -o errexit`, running the command on its own line would cause the script to exit if the command failed.

Reply:
Got it, thanks.